### PR TITLE
Add T1-FreeSurfer-Longitudinal pipeline

### LINKED ISSUE #121: t1-freesurfer-longitudinal-correction (recon-all -long) crashes when $SUBJECTS_DIR path is too long on macOS
Hello,

When running `t1-freesurfer-longitudinal-correction` with a long path for the working directory (and a result, the `SUBJECTS_DIR`), `recon-all -long` crashes without any detail (extract of `recon-all.log`):

```text
#-----------------------------------------
#@# Cortical Parc 2 lh Mon Aug 31 22:34:05 CEST 2020
/Users/alexandre.routier/FreeSurferLongitudinalCorrectionData/WorkingDirectoryWithVeryLongPath/t1-freesurfer-longitudinal-correction/ReconAll/sub-01_ses-2011_long-20112015/sub-01_ses-2011.long.sub-01_long-20112015/scripts
\n mris_ca_label -l ../label/lh.cortex.label -aseg ../mri/aseg.presurf.mgz -seed 1234 -long -R /Users/alexandre.routier/FreeSurferLongitudinalCorrectionData/WorkingDirectoryWithVeryLongPath/t1-freesurfer-longitudinal-correction/ReconAll/sub-01_ses-2011_long-20112015/sub-01_long-20112015/label/lh.aparc.a2009s.annot sub-01_ses-2011.long.sub-01_long-20112015 lh ../surf/lh.sphere.reg /Applications/Neuroimagerie/freesurfer/6.0.0/average/lh.CDaparc.atlas.acfb40.noaparc.i12.2016-08-02.gcs ../label/lh.aparc.a2009s.annot \n
setting seed for random number generator to 1234
using ../mri/aseg.presurf.mgz aseg volume to correct midline
will refine the initial labeling read-in from -R
reading precomputed parcellation from /Users/alexandre.routier/FreeSurferLongitudinalCorrectionData/WorkingDirectoryWithVeryLongPath/t1-freesurfer-longitudinal-correction/ReconAll/sub-01_ses-2011_long-20112015/sub-01_long-20112015/label/lh.aparc.a2009s.annot...
$Id: mris_ca_label.c,v 1.37 2014/02/04 17:46:42 fischl Exp $
  $Id: mrisurf.c,v 1.781.2.6 2016/12/27 16:47:14 zkaufman Exp $
reading atlas from /Applications/Neuroimagerie/freesurfer/6.0.0/average/lh.CDaparc.atlas.acfb40.noaparc.i12.2016-08-02.gcs...
reading color table from GCSA file....
average std = 2.9   using min determinant for regularization = 0.086
0 singular and 762 ill-conditioned covariance matrices regularized
input 0: MEAN CURVATURE, flags 0, avgs 5, name mean_curvature
Abort
Darwin UMR-COGIM-MP003-2.local 18.7.0 Darwin Kernel Version 18.7.0: Thu Jun 18 20:50:10 PDT 2020; root:xnu-4903.278.43~1/RELEASE_X86_64 x86_64

recon-all -s sub-01_ses-2011.long.sub-01_long-20112015 exited with ERRORS at Mon Aug 31 22:34:11 CEST 2020

To report a problem, see http://surfer.nmr.mgh.harvard.edu/fswiki/BugReporting
```
When a short path is given (e.g. `/tmp/fs`), it works like a charm.

I noticed that this was not the case on Ubuntu.

### PR DESCRIPTION
**Pull request by alexandreroutier**
Resolves #121

Normally, the CI should pass 🤞

### CODE REVIEW

File: clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_longitudinal_correction_utils.py [hunk 1]
```diff
@@ -5,14 +5,28 @@ def init_input_node(caps_dir, participant_id, session_id, long_id, output_dir):
...
+    import datetime
+    import platform
+    from tempfile import mkdtemp
+    from colorama import Fore
     from clinica.utils.longitudinal import read_sessions
     from clinica.utils.ux import print_begin_image
+    from clinica.utils.stream import cprint
 
     # Extract <image_id>
     image_id = '{0}_{1}_{2}'.format(participant_id, session_id, long_id)
```
Comment (mdiazmel) [inline clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_longitudinal_correction_utils.py:17]: This doesn't concern your changes, but now that we are updated with python >= 3.6, we should start changing all the "string format" by _f-strings_. E.g."
```suggestion
    image_id = f'{participant_id}_{session_id}_{long_id}'.
``

**Review by mdiazmel — COMMENTED** [review]
Hi @alexandreroutier ,
It seems ok! 
I suppose that adding this test will increase the duration of the non-regression tests. Do you have an idea about how long it's going to take running this single test (in our CI instances)?

**Comment by alexandreroutier** [comment]
> Hi @alexandreroutier ,
> It seems ok!
> I suppose that adding this test will increase the duration of the non-regression tests. Do you have an idea about how long it's going to take running this single test (in our CI instances)?

On average, it adds 2x 6-8 hours for `t1-freesurfer-template` and 4-6 hours for `t1-freesurfer-longitudinal-correction` so it is quite time consuming :(

**Comment by mdiazmel** [comment]
> 
> On average, it adds 2x 6-8 hours for `t1-freesurfer-template` and 4-6 hours for `t1-freesurfer-longitudinal-correction` so it is quite time consuming :(

How about isolating this test and run it only if the pipeline is modified ? (don't know how to do it exactly but we can maybe put it on the backlog...)

**Comment by alexandreroutier** [comment]
> > On average, it adds 2x 6-8 hours for `t1-freesurfer-template` and 4-6 hours for `t1-freesurfer-longitudinal-correction` so it is quite time consuming :(
> 
> How about isolating this test and run it only if the pipeline is modified ? (don't know how to do it exactly but we can maybe put it on the backlog...)

Maybe we can consider "daily" and  "weekly" non regression tests 🤔 

`t1-freesurfer*` are the only pipelines that exceed 2 hours of non regression tests. If we combine this with some UT for `t1_freesurfer_*utils.py` functions, we can be quite sure that we avoid issues.

**Comment by mdiazmel** [comment]
I'll merge this PR!

**Pull request merged**